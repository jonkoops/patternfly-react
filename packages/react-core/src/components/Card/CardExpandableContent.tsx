import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/Card/card';
import { css } from '@patternfly/react-styles';
import { CardContext } from './Card';

export interface CardExpandableContentProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the Card Body */
  children?: ReactNode;
  /** Additional classes added to the Card Body */
  className?: string;
}

export const CardExpandableContent: FunctionComponent<CardExpandableContentProps> = ({
  children,
  className,
  ...props
}: CardExpandableContentProps) => (
  <CardContext.Consumer>
    {({ isExpanded }) =>
      isExpanded ? (
        <div className={css(styles.cardExpandableContent, className)} {...props}>
          {children}
        </div>
      ) : null
    }
  </CardContext.Consumer>
);
CardExpandableContent.displayName = 'CardExpandableContent';
