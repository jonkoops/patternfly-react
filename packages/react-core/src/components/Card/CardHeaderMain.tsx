import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Card/card';

export interface CardHeaderMainProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the Card Head Main */
  children?: ReactNode;
  /** Additional classes added to the Card Head Main */
  className?: string;
}

export const CardHeaderMain: FunctionComponent<CardHeaderMainProps> = ({
  children,
  className,
  ...props
}: CardHeaderMainProps) => (
  <div className={css(styles.cardHeaderMain, className)} {...props}>
    {children}
  </div>
);
CardHeaderMain.displayName = 'CardHeaderMain';
