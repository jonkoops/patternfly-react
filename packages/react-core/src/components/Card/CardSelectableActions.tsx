import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Card/card';

export interface CardActionsProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the card action */
  children?: ReactNode;
  /** Additional classes added to the action */
  className?: string;
}

export const CardSelectableActions: FunctionComponent<CardActionsProps> = ({
  children,
  className,
  ...props
}: CardActionsProps) => (
  <div className={css(styles.cardSelectableActions, className)} {...props}>
    {children}
  </div>
);
CardSelectableActions.displayName = 'CardSelectableActions';
