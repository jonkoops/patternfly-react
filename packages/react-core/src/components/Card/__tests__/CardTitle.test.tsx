/**
 * @vitest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, test, vi } from 'vitest';

import { CardContext } from '../Card';
import { CardTitle } from '../CardTitle';

describe('CardTitle', () => {
  test('renders with PatternFly Core styles', () => {
    const { asFragment } = render(<CardTitle>text</CardTitle>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<CardTitle className="extra-class">text</CardTitle>);
    expect(screen.getByText('text')).toHaveClass('extra-class');
  });

  test('extra props are spread to the root element', () => {
    const testId = 'card-header';

    render(<CardTitle data-testid={testId} />);
    expect(screen.getByTestId(testId)).toBeInTheDocument();
  });

  test('calls the registerTitleId function provided by the CardContext with the generated title id', () => {
    const mockRegisterTitleId = vi.fn();

    render(
      <CardContext.Provider value={{ cardId: 'card', registerTitleId: mockRegisterTitleId }}>
        <CardTitle>text</CardTitle>
      </CardContext.Provider>
    );

    expect(mockRegisterTitleId).toHaveBeenCalledWith('card-title');
  });
});
