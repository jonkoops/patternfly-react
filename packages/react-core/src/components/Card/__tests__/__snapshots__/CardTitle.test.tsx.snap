// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardTitle > renders with PatternFly Core styles 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-card__title"
  >
    <div
      class="pf-v5-c-card__title-text"
    >
      text
    </div>
  </div>
</DocumentFragment>
`;
